src/orquestra/sdk/schema/ir.py

```diff
@@ -414,3 +417,41 @@ class WorkflowDef(BaseModel):
     # The resources that are available for the workflow to use.
     # If none, the runtime will decide.
     resources: t.Optional[Resources] = None
+
+    @pydantic.validator("metadata", always=True)
+    def sdk_version_up_to_date(cls, v: t.Optional[WorkflowMetadata]):
+        # Workaround for circular imports
+        from orquestra.sdk import exceptions
+        from orquestra.sdk.packaging import _versions
+
+        current_version = _versions.get_current_sdk_version()
+
+        if v is None:
+            warnings.warn(
+                exceptions.VersionMismatch(
+                    (
+                        "Attempting to read a workflow definition generated with an "
+                        "old version of Orquestra Workflow SDK. Please consider "
+                        "re-running your workflow or downgrading orquestra-sdk."
+                    ),
+                    actual=current_version,
+                    needed=None,
+                )
+            )
+            return v
+
+        if v.sdk_version != current_version:
+            warnings.warn(
+                exceptions.VersionMismatch(
+                    (
+                        "Attempting to read a workflow definition generated with a "
+                        "different version of Orquestra Workflow SDK. "
+                        "Please consider re-running your workflow or installing "
+                        f"'orquestra-sdk=={v.sdk_version.original}'"
+                    ),
+                    actual=current_version,
+                    needed=v.sdk_version,
+                )
+            )
+
+        return v
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**suggestion:** ‏I think this makes sense to be a semver check vs an entire version check.
Right now, because we're pre-1.0.0 it would give the same result (all pre v1 minor versions are technically semver breaking changes). 
My reasoninfg is that this would give us a clear line for "IR breakages are a major version bump" and would mean we don't bother people with warnings a lot between v1 and v2.

Reply:
It shouldn't take long and we'd just line ourselves up for tech debt.

```
if (current_version.major > 0 and current_version.major != v.sdk_version.major) or (current_version.major == 0 and current_version != v.sdk_version):
    pass
```

Reply:
Added in [d106280](https://github.com/zapatacomputing/orquestra-workflow-sdk/pull/105/commits/d10628093c867f02c2a724c3359a246e3fc5aa7f). WDYT?